File: code/209. Minimum Size Subarray Sum.cpp
class Solution {
public:
    int minSubArrayLen(int s, vector<int>& nums) {
        int ans = INT_MAX, cur = 0, tot = 0, left = 0, right = 0;
        while (right < nums.size()) {
            while (right < nums.size() && tot < s) {
                tot += nums[right];
                right++;
            } 
            while (left < right && tot >= s) {
                ans = min(ans, right - left);
                tot -= nums[left];
                left++;
            }
        }
        return ans == INT_MAX ? 0 : ans;
    }
};

File: code/209. Minimum Size Subarray Sum.go
func minSubArrayLen(s int, nums []int) int {
    left := 0
    ans := 1 << 30
    sum := 0
    for right := 0; right < len(nums); right++ {
        sum += nums[right]
        for sum >= s {
            if ans > right - left + 1 {
                ans = right - left + 1
            }
            sum -= nums[left]
            left++
        }
    }
    if ans == 1 << 30 {
        return 0
    } else {
        return ans
    }
}